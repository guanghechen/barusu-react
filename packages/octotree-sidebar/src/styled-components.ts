import type { OctotreeSidebarTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    octotreeSidebar?: OctotreeSidebarTheme
  }
}
