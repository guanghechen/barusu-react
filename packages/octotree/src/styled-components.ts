import type { OctotreeTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    octotree?: OctotreeTheme
  }
}
