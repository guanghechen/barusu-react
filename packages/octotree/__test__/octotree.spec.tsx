import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Octotree, { RawOctotreeNodeData, resolveOctotreeData } from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  const rawData: RawOctotreeNodeData[] = [
    {
      title: 'hooks',
      children: [
        {
          title: 'demo',
          pathname: '/hooks/demo',
          children: [{ title: '2', pathname: '/hooks/demo/2' }]
        },
        { title: 'ref', pathname: '/hooks/ref' }
      ]
    },
    { title: 'context', pathname: '/context' },
    { title: 'demo', pathname: '/demo' },
  ]

  const data = resolveOctotreeData('/api', rawData)

  it('nodes is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => render(
        <Router>
          <Octotree nodes={ value } />
        </Router>
      ))
        .toThrow(/Failed prop type: The prop `nodes` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const wrapper = mount(
      <Router>
        <Octotree ref={ ref } data-value="waw" nodes={ data } />
      </Router>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Router>
        <Octotree
          nodes={ data }
          iconWidth="1.5rem"
          style={{ color: 'orange', fontSize: '16px' }}
        />
      </Router>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      octotree: {
        fontSize: '18px',
        colorTitle: '#ccc',
        linkBackgroundHover: 'red',
        linkBackgroundActive: 'blue',
        typeIconColorSecondary: 'green'
      }
    }

    const wrapper = render(
      <ThemeProvider theme={ theme }>
        <Router>
          <Octotree
            nodes={ data }
            iconWidth="1.5rem"
            style={{ color: 'orange', fontSize: '16px' }}
          />
        </Router>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
