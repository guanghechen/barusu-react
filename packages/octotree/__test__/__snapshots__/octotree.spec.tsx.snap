// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c5 {
  display: inline-block;
  font-size: 1rem;
  font-family: octotree;
  font-style: normal;
  text-align: center;
}

.c5::before {
  font-size: inherit;
  font-style: normal;
}

.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  color: #0f2e47;
  font-weight: normal;
  font-style: normal;
}

.c12 {
  color: #8a8a8a;
}

.c12::before {
  content: "\\e827";
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  font-size: inherit;
  white-space: nowrap;
  background: none;
}

.c11:hover {
  background: #f6f8fa;
}

.c6::before {
  content: '\\e808';
}

.c7 {
  color: #6cb5fe;
}

.c7::before {
  content: '\\e840';
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c3:hover {
  background: #f6f8fa;
}

.c10 {
  display: block;
  overflow: auto;
  height: auto;
  opacity: 1;
}

.c2 .c4::before {
  content: '\\e80a';
}

.c2 .c9 {
  height: 0;
  opacity: 0;
}

.c1 {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
}

.c1 ul,
.c1 li {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
  line-height: 1.5;
  font-size: 1rem;
}

.c1 a {
  color: #0f2e47;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  text-indent: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  style="color:orange;font-size:16px"
>
  <ul
    class="c1"
  >
    <li>
      <div
        class="c2"
      >
        <div
          class="c3"
          style="padding-left:0"
        >
          <i
            class="c4 c5 c6"
          />
          <i
            class="c5 c7"
            style="padding-right:0.30rem"
          />
          <h4
            class="c8"
            title="hooks"
          >
            hooks
          </h4>
        </div>
        <div
          class="c9 c10"
        >
          <ul>
            <li>
              <div
                class="c2"
              >
                <div
                  class="c3"
                  style="padding-left:1.50rem"
                >
                  <i
                    class="c4 c5 c6"
                  />
                  <i
                    class="c5 c7"
                    style="padding-right:0.30rem"
                  />
                  <h4
                    class="c8"
                    title="demo"
                  >
                    demo
                  </h4>
                </div>
                <div
                  class="c9 c10"
                >
                  <ul>
                    <li>
                      <a
                        class="c11"
                        href="/api/hooks/demo/2"
                        style="padding-left:4.50rem"
                      >
                        <i
                          class="c5 c12"
                          style="padding-right:0.30rem"
                        />
                        <h4
                          class="c8"
                          title="2"
                        >
                          2
                        </h4>
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li>
              <a
                class="c11"
                href="/api/hooks/ref"
                style="padding-left:3.00rem"
              >
                <i
                  class="c5 c12"
                  style="padding-right:0.30rem"
                />
                <h4
                  class="c8"
                  title="ref"
                >
                  ref
                </h4>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <a
        class="c11"
        href="/api/context"
        style="padding-left:1.50rem"
      >
        <i
          class="c5 c12"
          style="padding-right:0.30rem"
        />
        <h4
          class="c8"
          title="context"
        >
          context
        </h4>
      </a>
    </li>
    <li>
      <a
        class="c11"
        href="/api/demo"
        style="padding-left:1.50rem"
      >
        <i
          class="c5 c12"
          style="padding-right:0.30rem"
        />
        <h4
          class="c8"
          title="demo"
        >
          demo
        </h4>
      </a>
    </li>
  </ul>
</div>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c5 {
  display: inline-block;
  font-size: 18px;
  font-family: octotree;
  font-style: normal;
  text-align: center;
}

.c5::before {
  font-size: inherit;
  font-style: normal;
}

.c8 {
  overflow: hidden;
  text-overflow: ellipsis;
  color: #ccc;
  font-weight: normal;
  font-style: normal;
}

.c12 {
  color: green;
}

.c12::before {
  content: "\\e827";
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  font-size: inherit;
  white-space: nowrap;
  background: none;
}

.c11:hover {
  background: red;
}

.c6::before {
  content: '\\e808';
}

.c7 {
  color: #6cb5fe;
}

.c7::before {
  content: '\\e840';
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c3:hover {
  background: red;
}

.c10 {
  display: block;
  overflow: auto;
  height: auto;
  opacity: 1;
}

.c2 .c4::before {
  content: '\\e80a';
}

.c2 .c9 {
  height: 0;
  opacity: 0;
}

.c1 {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
}

.c1 ul,
.c1 li {
  display: block;
  padding: 0;
  margin: 0;
  list-style: none;
  line-height: 1.5;
  font-size: 18px;
}

.c1 a {
  color: #ccc;
  outline: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  text-indent: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div
  class="c0"
  style="color:orange;font-size:16px"
>
  <ul
    class="c1"
  >
    <li>
      <div
        class="c2"
      >
        <div
          class="c3"
          style="padding-left:0"
        >
          <i
            class="c4 c5 c6"
          />
          <i
            class="c5 c7"
            style="padding-right:0.30rem"
          />
          <h4
            class="c8"
            title="hooks"
          >
            hooks
          </h4>
        </div>
        <div
          class="c9 c10"
        >
          <ul>
            <li>
              <div
                class="c2"
              >
                <div
                  class="c3"
                  style="padding-left:1.50rem"
                >
                  <i
                    class="c4 c5 c6"
                  />
                  <i
                    class="c5 c7"
                    style="padding-right:0.30rem"
                  />
                  <h4
                    class="c8"
                    title="demo"
                  >
                    demo
                  </h4>
                </div>
                <div
                  class="c9 c10"
                >
                  <ul>
                    <li>
                      <a
                        class="c11"
                        href="/api/hooks/demo/2"
                        style="padding-left:4.50rem"
                      >
                        <i
                          class="c5 c12"
                          style="padding-right:0.30rem"
                        />
                        <h4
                          class="c8"
                          title="2"
                        >
                          2
                        </h4>
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li>
              <a
                class="c11"
                href="/api/hooks/ref"
                style="padding-left:3.00rem"
              >
                <i
                  class="c5 c12"
                  style="padding-right:0.30rem"
                />
                <h4
                  class="c8"
                  title="ref"
                >
                  ref
                </h4>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </li>
    <li>
      <a
        class="c11"
        href="/api/context"
        style="padding-left:1.50rem"
      >
        <i
          class="c5 c12"
          style="padding-right:0.30rem"
        />
        <h4
          class="c8"
          title="context"
        >
          context
        </h4>
      </a>
    </li>
    <li>
      <a
        class="c11"
        href="/api/demo"
        style="padding-left:1.50rem"
      >
        <i
          class="c5 c12"
          style="padding-right:0.30rem"
        />
        <h4
          class="c8"
          title="demo"
        >
          demo
        </h4>
      </a>
    </li>
  </ul>
</div>
`;
